packages/css-engine/src/core/find-applicable-media.test.ts

```diff
@@ -0,0 +1,43 @@
+import { describe, test, expect } from "@jest/globals";
+import { findApplicableMedia } from "./find-applicable-media";
+
+const media = [
+  {},
+  { maxWidth: 991 },
+  { maxWidth: 767 },
+  { maxWidth: 479 },
+  { minWidth: 1280 },
+  { minWidth: 1440 },
+  { minWidth: 1920 },
+];
```

Comment:
What if maxWidth and minWidth would have intersections? How grouping etc would work too